app/src/features/rules/components/RulesList/components/RulesTable/hooks/useRuleTableColumns.tsx

```diff
@@ -140,12 +142,16 @@ const useRuleTableColumns = (options: Record<string, boolean>) => {
       width: 120,
       render: (_, record: RuleTableDataType, index) => {
         const isRecordActive = record.status === RecordStatus.ACTIVE;
-        // todo @nsr: check if this check also applies to groups?
         if (isRule(record)) {
           return (
             <PremiumFeature
               disabled={isRecordActive}
-              features={[FeatureLimitType.num_active_rules]}
+              features={
+                PREMIUM_RULE_TYPES.includes(record.ruleType)
+                  ? [FeatureLimitType.num_active_rules, FeatureLimitType.response_rule]
+                  : [FeatureLimitType.num_active_rules]
```

Comment:
Can this check be done by PremiumFeatures component?
only ruleType to be passed and check if its a premium rule to be done by PremiumFeatures component

Reply:
PremiumFeatures component is a HOC which wraps multiple premium features touchpoints like dynamic request/response body, rule sharing etc.We shouldn't write any feature specific logic inside this HOC